edxval/api.py

```diff
@@ -831,37 +823,70 @@ def export_to_xml(video_ids, course_id=None, external=False):
 
-    return create_transcripts_xml(video_ids, video_el)
 
+def create_trancript_file(edx_video_id, language_code, file_format, transcript_dir, resource_fs):
+    """
+    Writes transcript file to file system.
 
-def create_transcripts_xml(video_ids, video_el):
+    Arguments:
+        edx_video_id (str): Video id of the video transcript file is attached.
+        language_code (str): Language code of the transcript.
+        file_format (str): File format of the transcript file.
+        transcript_dir (str): The Directory to store transcript file.
+        resource_fs (SubFS): The file system to store transcripts.
+    """
+    if transcript_dir and resource_fs:
+        transcript_name = u'{transcript_dir}/{video_id}-{language_code}.{file_format}'.format(
+            transcript_dir=transcript_dir,
+            video_id=edx_video_id,
+            language_code=language_code,
+            file_format=file_format
+        )
+        transcript_data = get_video_transcript_data(edx_video_id, language_code)
+        if transcript_data:
+            transcript_content = transcript_data['content']
+            with resource_fs.open(transcript_name, 'wb') as f:
+                f.write(transcript_content)
+
+
+def create_transcripts_xml(video_id, video_el, resource_fs=None, transcript_dir=None):
     """
     Create xml for transcripts.
 
     Arguments:
-        video_ids (list): It can contain edx_video_id and/or multiple external video ids
+        video_id (str): Video id of the video.
         video_el (Element): lxml Element object
+        transcript_dir (str): The Directory to store transcript file.
+        resource_fs (PyFilesystem): The file system to store transcripts.
 
     Returns:
         lxml Element object with transcripts information
     """
-    video_transcripts = VideoTranscript.objects.filter(video__edx_video_id__in=video_ids).order_by('language_code')
+    video_transcripts = VideoTranscript.objects.filter(video__edx_video_id=video_id).order_by('language_code')
     # create transcripts node only when we have transcripts for a video
     if video_transcripts.exists():
         transcripts_el = SubElement(video_el, 'transcripts')
 
     exported_language_codes = []
     for video_transcript in video_transcripts:
         if video_transcript.language_code not in exported_language_codes:
+            edx_video_id = video_transcript.video.edx_video_id
+            language_code = video_transcript.language_code
+            transcript_url = video_transcript.transcript.url
+            file_format = video_transcript.file_format
+
+            create_trancript_file(edx_video_id, language_code, file_format, transcript_dir, resource_fs)
+
             SubElement(
                 transcripts_el,
                 'transcript',
                 {
-                    'video_id': video_transcript.video.edx_video_id,
+                    'video_id': edx_video_id,
                     'file_name': video_transcript.transcript.name,
```

Comment:
we don't need this.

Reply:
we can only send language_code, file_format, and  provider (video id can be retrieved from the component)